src/native/corehost/hostfxr.h

```diff
@@ -322,6 +323,10 @@ struct hostfxr_dotnet_environment_info
     const struct hostfxr_dotnet_environment_framework_info* frameworks;
 };
 
+typedef void(HOSTFXR_CALLTYPE* hostfxr_get_dotnet_environment_info_result_fn)(
+    const struct hostfxr_dotnet_environment_info* info,
+    void* result_context);
```

Comment:
Had to move it after the `struct hostfxr_dotnet_environment_framework_info` is defined, to fix:

```
  In file included from /Users/am11/projects/runtime5/src/native/corehost/test/mockhostfxr/test_c_api.c:8:
  /Users/am11/projects/runtime5/src/native/corehost/hostfxr.h:91:18: error: declaration of 'struct hostfxr_dotnet_environment_info' will not be visible outside of this function [-Werror,-Wvisibility]
     91 |     const struct hostfxr_dotnet_environment_info* info,
        |                  ^
  1 error generated.
  make[2]: *** [test/mockhostfxr/CMakeFiles/mockhostfxr_2_2.dir/test_c_api.c.o] Error 1
  make[2]: *** Waiting for unfinished jobs....
```